qa/tasks/cephfs/mount.py

```diff
@@ -59,24 +65,59 @@ def setupfs(self, name=None):
         self.fs.wait_for_daemons()
         log.info('Ready to start {}...'.format(type(self).__name__))
 
-    def mount(self, mount_path=None, mount_fs_name=None, mountpoint=None, mount_options=[]):
+    def mount(self, client_id=None, client_keyring_path=None,
+              client_remote=None, cephfs_name=None, cephfs_mntpt=None,
+              hostfs_mntpt=None, mntopts=[], createfs=True):
         raise NotImplementedError()
 
     def umount(self):
         raise NotImplementedError()
 
-    def umount_wait(self, force=False, require_clean=False):
+    def umount_wait(self, force=False, require_clean=False, timeout=None,
+                    cleanup=True):
         """
 
         :param force: Expect that the mount will not shutdown cleanly: kill
                       it hard.
         :param require_clean: Wait for the Ceph client associated with the
                               mount (e.g. ceph-fuse) to terminate, and
                               raise if it doesn't do so cleanly.
+        :param timeout: amount of time to be waited for umount command to finish
+        :param cleanup: whether to delete mount point or not.
         :return:
         """
         raise NotImplementedError()
 
+    def remount(self, client_id=None, client_keyring_path=None, client_remote=None,
```

Comment:
I found this more natural when I wrote it since I was thinking remounting with different credentials. But creating a new mount object actually might be better...